src/main/java/concurrency/second/executor/ExecutionService.java

```diff
@@ -0,0 +1,42 @@
+package concurrency.second.executor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.apache.commons.collections4.ListUtils;
+
+public class ExecutionService {
+    public static final int THEADS = 12;
+    private List<Integer> list;
+
+    public ExecutionService(List<Integer> list) {
+        this.list = list;
+    }
+
+    public List<Integer> getList() {
+        return list;
+    }
+
+    public Integer execute(List<Integer> list) {
+        List<List<Integer>> partition = ListUtils.partition(list, list.size() / THEADS);
+        List<ClassImplementsCallable> callableList = new ArrayList<>();
```

Comment:
```suggestion
        List<Callable> tasks = new ArrayList<>();
```

Reply:
don't work with this changes

Reply:
user List<Callable<Integer> tasks
List<Future<Integer>> results = ....invokeAll(tasks);
then in for each iterate results with get method 
this way it'll work